packages/@react-aria/focus/test/FocusScope.test.js

```diff
@@ -595,6 +595,57 @@ describe('FocusScope', function () {
       userEvent.tab();
       expect(document.activeElement).toBe(getByTestId('after'));
     });
+
+    describe('focusable first in scope', function () {
+      beforeEach(() => {
+        jest.useFakeTimers();
+      });
+      afterEach(() => {
+        jest.runAllTimers();
+        jest.useRealTimers();
+      });
+      it('should restore focus to the first focusable or tabbable element within the scope when focus is lost within the scope', async function () {
+        let {getByTestId} = render(
+          <div>
+            <FocusScope contain>
+              <div role="dialog" data-testid="focusable" tabIndex={-1}>
+                <Item data-testid="tabbable1" autoFocus tabIndex={null}>Remove me!</Item>
+                <Item data-testid="item1" tabIndex={0}>Remove me, too!</Item>
+                <Item data-testid="item2" tabIndex={-1}>Remove me, three!</Item>
+              </div>
+            </FocusScope>
+          </div>
+        );
+
+        function Item(props) {
+          let focusManager = useFocusManager();
+          let onClick = e => {
+            focusManager.focusNext();
+            act(() => {
+              // remove fails to fire blur event in jest-dom
+              e.target.blur();
+              e.target.remove();
+              jest.runAllTimers();
+            });
```

Comment:
Hrm, using `focusManager.focusNext` means the test is manually focusing the correct button rather than relying on FocusScope's `onBlur` calling `focusFirstInScope` right? 

Don't wanna be too nitpicky with the test and this does test that `focusFirstInScope` gets called onBlur when item2 gets removed so I'm fine with its current state but it would be nice to verify that `focusFirstInScope` prioritizes the tabbable elements in a test. No need to spend too long on it if it is difficult to mock jest-dom in just the right way, can punt for later

Reply:
Without the `focusManager.focusNext`, the focus goes to the `document.body` in the testing environment. I've updated the Storybook story to better demonstrate that removing the last tabbable descendant from the FocusScope will focus the first focusable descendant within the scope.